### PR TITLE
Test in-progress builds.

### PR DESCRIPTION
**Pull request by mitchell-as**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2698" title="DX-2698" target="_blank">TaskDX-2698</a>  We have integration tests for in-progress builds
  </td></table>
  <br />

### CODE REVIEW

File: internal/runbits/runtime/refresh.go [hunk 1]
```diff
@@ -1,6 +1,8 @@
 package runtime
```
Comment (mitchell-as) [inline internal/runbits/runtime/refresh.go:1]: Note: the changes in this file are in #3332.

File: test/integration/package_int_test.go [hunk 2]
```diff
@@ -783,6 +784,56 @@ func (suite *PackageIntegrationTestSuite) TestChangeSummary() {
...
+	cp = ts.Spawn("config", "set", constants.AsyncRuntimeConfig, "true")
+	cp.Expect("Successfully set")
+	cp.ExpectExitCode(0)
+
+	cp = ts.Spawn("checkout", "ActiveState-CLI/Perl-5.36", ".")
+	cp.Expect("Checked out")
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnWithOpts(
+		e2e.OptArgs("install", "private/"+e2e.PersistentUsername+"/hello-world", "--ts", "now"),
```
Comment (mitchell-as) [inline test/integration/package_int_test.go:820]: The JIRA ticket says to not use `--ts now`, but I think that is based on outdated behavior. I need `--ts now` to install the newest version just published in the lines above.

File: test/integration/package_int_test.go [hunk 3]
```diff
@@ -783,6 +784,56 @@ func (suite *PackageIntegrationTestSuite) TestChangeSummary() {
...
+	)
+	cp.Expect("Package added: hello-world")
+	cp.ExpectExitCode(0)
+
+	cp = ts.SpawnWithOpts(
+		e2e.OptArgs("exec", "main"),
+		e2e.OptAppendEnv(constants.DisableRuntime+"=false"),
+	)
+	cp.Expect("Build Log")
+	cp.Expect("Building")
```
Comment (mitchell-as) [inline test/integration/package_int_test.go:831]: The JIRA ticket is scant on details as to what exactly to test, so I decided to look for this.

**Comment by mitchell-as** [comment]
Test failures are known issues.

File: internal/runbits/runtime/refresh.go (hunk 1 above)
Comment (mitchell-as) [reply internal/runbits/runtime/refresh.go:1]: Update: I've closed that PR. This PR will have these temporary changes until the runtime refactor mentioned in https://github.com/ActiveState/cli/pull/3332#issuecomment-2140882402 will fix this properly.

File: test/integration/package_int_test.go [hunk 4]
```diff
@@ -783,6 +784,56 @@ func (suite *PackageIntegrationTestSuite) TestChangeSummary() {
...
+		"--depend", "builder/autotools-builder@>=0", // for ./configure, make, make install
+		"--depend", "internal/mingw-build-selector@>=0", // for Windows to use mingw's GCC
+		filepath.Join(osutil.GetTestDataDir(), "hello-world-1.0.0.tar.gz"),
+		"--edit") // publish a new revision each time, forcing a build
+	cp.Expect("Successfully published")
+	cp.ExpectExitCode(0)
+
+	cp = ts.Spawn("config", "set", constants.AsyncRuntimeConfig, "true")
+	cp.Expect("Successfully set")
+	cp.ExpectExitCode(0)
```
Comment (Naatan) [inline test/integration/package_int_test.go:813]:
```suggestion
```

I don't think this should be in this test at all, unless you unset it again right before you run the install command. With this set there shouldn't be any build happening at all. The "fix" to refresh.go is just working around the problem you introduced here.

The checkout already has runtime disabled, and for install you specifically enable it again. You can drop this code and the changes to refresh.go and everything should work.

**Review by Naatan — CHANGES_REQUESTED** [review]

File: test/integration/package_int_test.go [hunk 5]
```diff
@@ -783,6 +784,56 @@ func (suite *PackageIntegrationTestSuite) TestChangeSummary() {
 	cp.ExpectExitCode(0)
 }
 
+func (suite *PackageIntegrationTestSuite) TestBuildInProgress() {
+	suite.OnlyRunForTags(tagsuite.Package)
```
Comment (Naatan) [inline test/integration/package_int_test.go:788]: Let's move this to `runtime_int_test.go`. I don't think package is appropriate, this isn't doing anything with `state packages`.

File: internal/runbits/runtime/refresh.go (hunk 1 above)
Comment (mitchell-as) [reply internal/runbits/runtime/refresh.go]: Based on subsequent comments, this no longer applies.

File: test/integration/package_int_test.go (hunk 4 above)
Comment (mitchell-as) [reply test/integration/package_int_test.go]: Yep, that shows just how little I understand about the async mechanic. Based on your feedback I got it to work.

**Review by Naatan — APPROVED** [review]

**Pull request merged**